docs/development/pigs/pig-028.rst

```diff
@@ -0,0 +1,298 @@
+        class Config:
+            arbitrary_types_allowed = True
+            validate_all = True
+            validate_assignment = True
+            extra = "allow"
+
+        def to_header(self):
+            hdr_dict = {}
+            for key, item in self.dict().items():
+                hdr_dict[key.upper()] = item.__str__()
+            return hdr_dict
+
+        @classmethod
+        def from_header(cls, hdr):
+            kwargs = {}
+            for key in cls.__fields__.keys():
+                kwargs[key] = hdr.get(key.upper(), None)
+            return cls(**kwargs)
+
+The model `Config` defined allows:
+- using any type input and not only simple `Annotation` types (`arbitrary_types_allowed = True`)
+- Setting the `validate_assignment` to `True` ensures that validation is performed when a value
+  is assigned to the attribute.
+- `extra = "allow"` accepts additional attributes not defined in the metadata class.
+
+
+
+
+arbitrary type input and validation
+-----------------------------------
+
+By providing a validation method, it is possible to validate non-standard objects. The
+`validator` decorator provided by pydantic makes it easy. As shown below:
+
+.. code ::
+
+    class ArbitraryTypeMetaData(MetaDataBaseModel):
+        # allow string defining angle or Angle object
+        zenith_angle : Optional[Union[str, Angle]]
+        pointing_altaz : Union[]
+
+        # allow observatory name or astropy EarthLocation object
+        location : Optional[Union[str, EarthLocation]]
+
+        @validator('location')
+        def validate_location(cls, v):
+            if isinstance(v, str) and v in observatory_locations.keys():
+                return observatory_locations[v]
+            elif isinstance(v, EarthLocation):
+                return v
+            else:
+                raise ValueError("Incorrect location value")
+
+        @validator('zenith_angle')
+        def validate_zenith_angle(cls, v):
+            return Angle(v)
+
+
+Alternatives
+============
+
+Proposed metadata classes
+=========================
+
+Here we list the expected metadata classes that we expect. All classes will inherit from a
+parent ``MetaDataBase`` that will provide most base properties to the daughter classes.
+
+We provide the list of classes by subpackage
+
+data
+----
+
+- ``EventListMetaData``
+- ``ObservationMetaData``
+- ``DataStoreMetaData``
+- ``PointingMetaData``
+- ``GTIMetaData``
+
+IRF
+---
+
+Here we should distinguish between actual IRFs and reduced modeling-ready IRFs such as kernels
+and IRF maps.
+
+- ``IRFMetaData``
+  A single generic class could be used for all actual IRFs.
+
+Makers
+------
+
+It is unclear whether stateless algorithm classes such as ``Maker`` actually need meta
+information beyond their actual attributes. They will have to create or update `meta`
+information of the ``Dataset`` they create or modify. For now, we don't propose
+any metadata for ``Maker`` objects.
+
```

Comment:
I think that it might be important to store the options of the makers (e.g. fit or scale for the FoVBkgMaker).
However, these information could be classified as provenance information... or not. Maybe this is a point to discuss.